BoneReconstructionPlanner/BoneReconstructionPlanner.py

```diff
@@ -553,120 +597,161 @@ def process(self,fibulaLine,mandibularCurve,planeList,initialSpace,betweenSpace)
-      newPlane1.SetNormal(plane1Normal)
-      newPlane1.SetOrigin(lineStartPos)
-      newPlane2.SetNormal(plane2Normal)
-      newPlane2.SetOrigin(lineStartPos)
-
-      #Create origin1-origin2 versor
+      mandiblePlane0 = planeList[i]
+      mandiblePlane1 = planeList[i+1]
+      mandiblePlane0Normal = [0,0,0]
+      mandiblePlane0.GetNormal(mandiblePlane0Normal)
+      mandiblePlane1Normal = [0,0,0]
+      mandiblePlane1.GetNormal(mandiblePlane1Normal)
+      mandiblePlane0Origin = [0,0,0]
+      mandiblePlane0.GetOrigin(mandiblePlane0Origin)
+      mandiblePlane1Origin = [0,0,0]
+      mandiblePlane1.GetOrigin(mandiblePlane1Origin)
+      fibulaPlaneA = fibulaPlanesList[2*i]
+      fibulaPlaneB = fibulaPlanesList[2*i+1]
+      fibulaPlaneA.SetNormal(mandiblePlane0Normal)
+      fibulaPlaneA.SetOrigin(mandiblePlane0Origin)
+      fibulaPlaneB.SetNormal(mandiblePlane1Normal)
+      fibulaPlaneB.SetOrigin(mandiblePlane1Origin)
+
+      #Create origin1-origin2 vector
+      or0 = np.zeros(3)
       or1 = np.zeros(3)
-      or2 = np.zeros(3)
-      plane1.GetOrigin(or1)
-      plane2.GetOrigin(or2)
-      d.append(np.linalg.norm(or2-or1))
-      dVersor = (or2-or1)/np.linalg.norm(or2-or1)
+      mandiblePlane0.GetOrigin(or0)
+      mandiblePlane1.GetOrigin(or1)
+      boneSegmentsDistance.append(np.linalg.norm(or1-or0))
+      mandibleAxisZ = (or1-or0)/np.linalg.norm(or1-or0)
+      
+      #Get Y component of mandiblePlane0
+      mandiblePlane0matrix = vtk.vtkMatrix4x4()
+      mandiblePlane0.GetPlaneToWorldMatrix(mandiblePlane0matrix)
+      mandiblePlane0Y = np.array([mandiblePlane0matrix.GetElement(0,1),mandiblePlane0matrix.GetElement(1,1),mandiblePlane0matrix.GetElement(2,1)])
+      
+      mandibleAxisX = [0,0,0]
+      vtk.vtkMath.Cross(mandiblePlane0Y, mandibleAxisZ, mandibleAxisX)
+      mandibleAxisX = mandibleAxisX/np.linalg.norm(mandibleAxisX)
+      mandibleAxisY = [0,0,0]
+      vtk.vtkMath.Cross(mandibleAxisZ, mandibleAxisX, mandibleAxisY)
+      mandibleAxisY = mandibleAxisY/np.linalg.norm(mandibleAxisY)
 
       #Start transformations
       rotAxis = [0,0,0]
-      vtk.vtkMath.Cross(dVersor, lineDirectionVersor, rotAxis)
+      vtk.vtkMath.Cross(mandibleAxisZ, fibulaZ, rotAxis)
       rotAxis = rotAxis/np.linalg.norm(rotAxis)
-      angleRad = vtk.vtkMath.AngleBetweenVectors(dVersor, lineDirectionVersor)
+      angleRad = vtk.vtkMath.AngleBetweenVectors(mandibleAxisZ, fibulaZ)
       angleDeg = vtk.vtkMath.DegreesFromRadians(angleRad)
 
-      #this versor is created to check if rotAxis is okey or should be opposite
-      v1l = [0,0,0]
-      q = [angleRad,rotAxis[0],rotAxis[1],rotAxis[2]]
-      vtk.vtkMath.RotateVectorByWXYZ(dVersor,q,v1l)
+      #this vector is created to check if rotAxis is okey or should be opposite
+      rotatedMandibleAxisZ = [0,0,0]
+      rotation = [angleRad,rotAxis[0],rotAxis[1],rotAxis[2]]
+      vtk.vtkMath.RotateVectorByWXYZ(mandibleAxisZ,rotation,rotatedMandibleAxisZ)

```

Comment:
This is messy. Instead of RotateVectorByWXYZ use RotateX, RotateY, or RotateZ. We should almost never rotate along an arbitrary axis, because if there is any non-trivial direction then we specify a new coordinate system for that and will rotate along axis of that coordinate system.

Reply:
I think these variables won't be necessary after I remake the rotation transforms